Home Assistant not compatible with ffmpeg release 6

### The problem

It seeems  ha-av does not support ffmpeg6 yet. 
I have downgraded ffmpeg to 5.1 and it works again.

There are changes from ffmpeg 5.1 to ffmpeg 6 that prevents ha-av from compiling wheel...

Apr 06 19:48:05 HomeAssistant hass[4128]:       src/av/container/core.c:4966:118: note: expected ‘struct AVOutputFormat *’ but argument is of type ‘const struct AVOutputFormat *’
Apr 06 19:48:05 HomeAssistant hass[4128]:       src/av/container/core.c: In function ‘__pyx_pymod_exec_core’:
Apr 06 19:48:05 HomeAssistant hass[4128]:       src/av/container/core.c:9067:36: error: ‘AVFMT_FLAG_PRIV_OPT’ undeclared (first use in this function)
**Apr 06 19:48:05 HomeAssistant hass[4128]:        9067 |   __pyx_t_3 = __Pyx_PyInt_From_int(AVFMT_FLAG_PRIV_OPT); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 160, __pyx_L1_error)
Apr 06 19:48:05 HomeAssistant hass[4128]:             |                                    ^~~~~~~~~~~~~~~~~~~**
Apr 06 19:48:05 HomeAssistant hass[4128]:       src/av/container/core.c:9067:36: note: each undeclared identifier is reported only once for each function it appears in
Apr 06 19:48:05 HomeAssistant hass[4128]:       error: command '/usr/bin/gcc' failed with exit code 1
Apr 06 19:48:05 HomeAssistant hass[4128]:       [end of output]
Apr 06 19:48:05 HomeAssistant hass[4128]:
Apr 06 19:48:05 HomeAssistant hass[4128]:   note: This error originates from a subprocess, and is likely not a problem with pip.
Apr 06 19:48:05 HomeAssistant hass[4128]:   ERROR: Failed building wheel for ha-av
Apr 06 19:48:05 HomeAssistant hass[4128]: ERROR: Could not build wheels for ha-av, which is required to install pyproject.toml-based projects

ha-av is a fork of [PyAV](https://pypi.org/project/av/) which is used for providing hotfixes

PyAV will build from this commit https://github.com/wader/PyAV/commit/8871e13848cbf78615ba7b429d301adc071841ea

...so bringing ha-av up to speed with this commit may solve he problem...




### What version of Home Assistant Core has the issue?

2023,4,0

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

_No response_

### Link to integration documentation on our website

_No response_

### Diagnostics information

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
Apr 06 19:48:05 HomeAssistant hass[4128]:        9067 |   __pyx_t_3 = __Pyx_PyInt_From_int(AVFMT_FLAG_PRIV_OPT); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 160, __pyx_L1_error)
Apr 06 19:48:05 HomeAssistant hass[4128]:             |                                    ^~~~~~~~~~~~~~~~~~~
```


### Additional information

_No response_

--- a/homeassistant/components/generic/manifest.json
+++ b/homeassistant/components/generic/manifest.json
@@ -6,5 +6,5 @@
   "dependencies": ["http"],
   "documentation": "https://www.home-assistant.io/integrations/generic",
   "iot_class": "local_push",
-  "requirements": ["ha-av==10.0.0", "pillow==9.5.0"]
+  "requirements": ["ha-av==10.1.0", "pillow==9.5.0"]
 }

--- a/homeassistant/components/stream/manifest.json
+++ b/homeassistant/components/stream/manifest.json
@@ -7,5 +7,5 @@
   "integration_type": "system",
   "iot_class": "local_push",
   "quality_scale": "internal",
-  "requirements": ["PyTurboJPEG==1.6.7", "ha-av==10.0.0", "numpy==1.23.2"]
+  "requirements": ["PyTurboJPEG==1.6.7", "ha-av==10.1.0", "numpy==1.23.2"]
 }

--- a/homeassistant/package_constraints.txt
+++ b/homeassistant/package_constraints.txt
@@ -21,7 +21,7 @@ ciso8601==2.3.0
 cryptography==40.0.2
 dbus-fast==1.86.0
 fnv-hash-fast==0.3.1
-ha-av==10.0.0
+ha-av==10.1.0
 hass-nabucasa==0.67.1
 hassil==1.0.6
 home-assistant-bluetooth==1.10.0

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -869,7 +869,7 @@ h2==4.1.0
 
 # homeassistant.components.generic
 # homeassistant.components.stream
-ha-av==10.0.0
+ha-av==10.1.0
 
 # homeassistant.components.ffmpeg
 ha-ffmpeg==3.1.0

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -673,7 +673,7 @@ h2==4.1.0
 
 # homeassistant.components.generic
 # homeassistant.components.stream
-ha-av==10.0.0
+ha-av==10.1.0
 
 # homeassistant.components.ffmpeg
 ha-ffmpeg==3.1.0

